### PR TITLE
make BridgeBuilder a subclass of ThingBuilder and adapt BaseBridgeHandler.editThing()

### LINKED ISSUE #1887: BaseBridgeHandler.editThing() returns a ThingBuilder instead of a BridgeBuilder
Calling editThing() on a BaseBridgeHandler and updating the bridge returns a ThingImpl instead of a BridgeImpl. Edit operation should not convert a Bridge into a regular Thing.

This is similar to the issue resolved in #1467

### PR DESCRIPTION
**Pull request by kaikreuzer**
fixes #1887

Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

**Comment by maggu2810** [comment]
The usage of the method self() is now obsolete. Isn't it?

The with... functions return a ThingBuilder also for a BridgeBuilder. It is okay, as it is not possible (for me... I did not find a way to use a derived class).
Neverless, should we overwrite the build() method in the bridge handler that is returns a bridge?
Sure, if someone calls with... the type is hold as ThingBuilder and not as BridgeBuilder, but it will be good to document that build of a BridgeBuilder returns a bridge... WDYT?

**Comment by kaikreuzer** [comment]
I agree with both comments and updated the PR accordingly.

**Comment by kaikreuzer** [comment]
Just noticed that I missed a reference in model.thing... Will update the PR soon.

**Comment by kaikreuzer** [comment]
No, I was wrong - this was only a reference in the generated java file - the xtend version is clean. So it should all be fine.

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/BaseBridgeHandler.java [hunk 1]
```diff
@@ -53,4 +55,18 @@ public Thing getThingByUID(ThingUID uid) {
...
     }
+
+    /**
+     * Creates a bridge builder, which allows to modify the bridge. The method
+     * {@link BaseThingHandler#updateThing(Thing)} must be called to persist the changes.
+     *
+     * @return {@link BridgeBuilder} which builds an exact copy of the bridge (not null)
+     */
+    @Override
+    protected ThingBuilder editThing() {
```
Comment (maggu2810) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/BaseBridgeHandler.java]: One last comment: Should we return BridgeBuilder, here?
As BridgeBuilder is a child of ThingBuilder, I assume it works.

Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/BaseBridgeHandler.java]: like this?

**Comment by maggu2810** [comment]
Thanks

**Comment by kaikreuzer** [comment]
thank you for the detailed review!

**Pull request merged**